### DISCUSSION TITLE
Fragment Caching

### DISCUSSION THREAD

**Discussion by nimmolo** [MushroomObserver/mushroom-observer#1044] [Ideas]
Has anyone tried implementing [Rails native Fragment Caching](https://guides.rubyonrails.org/v6.1/caching_with_rails.html#fragment-caching) in the templates?

I'm going to try a PR against master right now -- please nominate your toppermost expensive partials here. shared/matrix_box and everything on show_observation seem like good places to start.

**Comment by pellaea** [comment]
I'm sorry to be negative, but almost nothing is going to work right now
because we customize the appearance of too much of the page.  Obviously
language is a factor, but also things like location order, whether to
display authorities or not, all sorts of controls depend on whether the
user is the owner or not, logged in or not, theme, etc. etc. etc.

I've discussed at length a number of times how we can refactor the
structure of our views to move all user-specific formatting into the
browser so that the content the server sends out can be cached.  I really
don't feel like writing it all up yet again.  No one ever listens to me!

In short: yes, I've looked at fragment caching.  It's a no go until we do
some major refactoring.

But prove me wrong.  I would love nothing better!

**Reply by nimmolo** [reply]
Ok. Let's try it! Are there potential downsides? We can always revert...

**Comment by pellaea** [comment]
I assume not... except the potential waste of your time!

**Reply by nimmolo** [reply]
That's ok... took 5 minutes. Fire when ready!
